src/journal/journal-file.c

```diff
@@ -464,6 +497,9 @@ static int journal_file_refresh_header(JournalFile *f) {
         /* Sync the online state to disk */
         (void) fsync(f->fd);
 
+        /* We likely just created a new file, also sync the directory this file is located in. */
+        (void) fsync_directory_of_file(f->fd);
```

Comment:
Ok, well I'm a little preoccupied with rebuilding a car right now, but after that's finished I'll likely be back to hacking on systemd stuff and I can poke at this then.